sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/ExtRefInfo.java

```diff
@@ -78,18 +82,15 @@ public static ExtRefInfo from(TExtRef tExtRef, String iedName, String ldInst,
      * @param tfcda FCDA data to check compatibilities with ExtRef
      * @return true if ExtRef matches FCDA for parameters ahead false otherwise
      */
-    public boolean checkMatchingFCDA(@NonNull TFCDA tfcda){
-        if(bindingInfo == null  || signalInfo == null) return false;
-        FCDAInfo fcdaInfo = new FCDAInfo(tfcda);
-        FCDAInfo fcdaOfBinding = FCDAInfo.builder()
-                .ldInst(bindingInfo.getLdInst())
-                .lnClass(bindingInfo.getLnClass())
-                .lnInst(bindingInfo.getLnInst())
-                .prefix(bindingInfo.getPrefix())
-                .doName(new DoTypeName(signalInfo.getPDO()))
-                .daName(new DaTypeName(signalInfo.getPDA()))
-                .build();
-        return fcdaInfo.checkFCDACompatibilitiesForBinding(fcdaOfBinding);
+    public boolean checkMatchingFCDA(@NonNull TFCDA tfcda) {
+        if (bindingInfo == null || signalInfo == null) return false;
+
+        return trimToEmpty(tfcda.getLdInst()).equals(trimToEmpty(bindingInfo.getLdInst()))
+                && trimToEmpty(tfcda.getPrefix()).equals(trimToEmpty(bindingInfo.getPrefix()))
+                && trimToEmpty(tfcda.getLnClass().get(0)).equals(trimToEmpty(bindingInfo.getLnClass()))
+                && trimToEmpty(tfcda.getLnInst()).equals(trimToEmpty(bindingInfo.getLnInst()))
```

Comment:
for this I would prefer to use a Fcda Object, because I think we should not mix many entities.
So `Fcda.from(myExtRefInfo).equals(myFcda)` is the exact purpose of this `checkMatchingFCDA` method